src/filter/median.rs

```diff
@@ -289,43 +322,50 @@ mod tests {
     use std::cmp::{min, max};
 
     macro_rules! bench_median_filter {
-        ($name:ident, side: $s:expr, radius: $r:expr) => {
+        ($name:ident, side: $s:expr, x_radius: $rx:expr, y_radius: $ry:expr) => {
             #[bench]
             fn $name(b: &mut Bencher) {
                 let image = gray_bench_image($s, $s);
                 b.iter(|| {
-                    let filtered = median_filter(&image, $r);
+                    let filtered = median_filter(&image, $rx, $ry);
                     black_box(filtered);
                 })
             }
         }
     }
 
-    bench_median_filter!(bench_median_filter_s100_r1, side: 100, radius: 1);
-    bench_median_filter!(bench_median_filter_s100_r4, side: 100, radius: 4);
-    bench_median_filter!(bench_median_filter_s100_r8, side: 100, radius: 8);
+    bench_median_filter!(bench_median_filter_s100_r1, side: 100, x_radius: 1,y_radius: 1);
+    bench_median_filter!(bench_median_filter_s100_r4, side: 100, x_radius: 4,y_radius: 4);
+    bench_median_filter!(bench_median_filter_s100_r8, side: 100, x_radius: 8,y_radius: 8);
+    
+    // benchmark on non-square kernels
+    bench_median_filter!(bench_median_filter_s100_rx1_ry4, side: 100, x_radius: 1,y_radius: 4);
+    bench_median_filter!(bench_median_filter_s100_rx1_ry8, side: 100, x_radius: 1,y_radius: 8);
+    bench_median_filter!(bench_median_filter_s100_rx4_ry8, side: 100, x_radius: 4,y_radius: 1);
+    bench_median_filter!(bench_median_filter_s100_rx8_ry1, side: 100, x_radius: 8,y_radius: 1);
+
 
-    // Reference implementation of median filter - written to be as simple as possible,
```

Comment:
Why did you remove this comment?

Reply:
Unintentionally deleted it, recover it in the new commit